docs/documentation/guides/event-filter.mdx

```diff
@@ -0,0 +1,503 @@
+  "price": 0,
+  "title": "15 Min Meeting between Eric Allam and John Doe",
+  "length": 15,
+  "status": "ACCEPTED",
+  "endTime": "2023-01-25T16:00:00Z",
+  "bookingId": 198052,
+  "organizer": {
+    "id": 32794,
+    "name": "Eric Allam",
+    "email": "eric@trigger.dev",
+    "language": { "locale": "en" },
+    "timeZone": "Europe/London"
+  }
+}
+```
+
+The following event filter would match the event:
+
+```json
+{
+  "type": ["15 Min Meeting"],
+  "status": ["ACCEPTED", "REJECTED"],
+  "organizer": {
+    "name": ["Eric Allam"]
+  }
+}
+```
+
+For an event pattern to match an event, the event must contain all the field names listed in the event pattern. The field names must also appear in the event with the same nesting structure.
+
+The value of each field name in the event pattern must be an array of strings, numbers, or booleans. The event pattern matches the event if the value of the field name in the event is equal to any of the values in the array.
+
+Effectively, each array is an OR condition, and the entire event pattern is an AND condition.
+
+So the above event filter will match because status == "ACCEPTED", and it would also match if status == "REJECTED".
+
+## Other types of triggers that can also support event filters.
+
+#### 1. Stripe Trigger with Event Filter
+
+**Here is an example of a Stripe Trigger with an Event Filter:**
+
+```ts
+client.defineJob({
+  id: "stripe-on-subscription-created",
+  name: "Stripe On Subscription Created",
+  version: "0.1.0",
+  trigger: stripe.onCustomerSubscriptionCreated({
+    filter: {
+      currency: ["usd"],
+    },
+  }),
+  run: async (payload, io, ctx) => {
+    await io.logger.info("ctx", { ctx });
+  },
+});
+```
+
+The job is triggered by the `onCustomerSubscriptionCreated` event from Stripe. It is configured to trigger only when certain conditions are met. In this case, the job is triggered when a subscription is created with the following condition:
+
+- Currency: Only subscriptions with the currency "USD" will trigger this job.
+
+#### 2. Supabase trigger with Event Filter
+
+**Here is an example of a Supabase Trigger with an Event Filter:**
+
+```ts
+client.defineJob({
+  id: "supabase-management-example-objects-storage",
+  name: "Supabase Management Example Object Storage",
+  version: "0.1.0",
+  trigger: triggers.onInserted({
+    schema: "storage",
+    table: "objects",
+    filter: {
+      record: {
+        bucket_id: ["example_bucket"], // Only trigger for objects in the "example_bucket" bucket
+        name: [
+          {
+            $endsWith: ".png", // Only trigger for objects with a name ending in ".png"
+          },
+        ],
+        path_tokens: [
+          {
+            $includes: "images", // Only trigger for objects with a path that includes the token "images"
+          },
+        ],
+      },
+    },
+  }),
+  integrations: {
+    openai,
+    supabase,
+  },
+  run: async (payload, io, ctx) => {
+    if (!payload.record.name) {
```

Comment:
Remove the body of the run function, try to keep the example jobs as simple as possible